src/main/java/com/programmers/springweekly/util/GeneratorDeepCopiedType.java

```diff
@@ -0,0 +1,20 @@
+package com.programmers.springweekly.util;
+
+import org.springframework.stereotype.Component;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class GeneratorDeepCopiedType <K, V>{
+
+    public Map<K, V> copiedMap(Map<K, V> map){
+        Map<K, V> copiedMap = new ConcurrentHashMap<>();
+
+        for(Map.Entry<K, V> entry : map.entrySet()){
+            copiedMap.put(entry.getKey(), entry.getValue());
+        }
+
+        return copiedMap;
+    }
+}
```

Comment:
우선 위 리뷰에서 깊은 복사하기 클래스가 필요 없다고 말씀드리긴 했는데, 제네릭 관련해서 말씀드리면..

클래스 선언부의 제네릭은 내부 멤버 모두가 사용 가능한 제네릭입니다.
하지만 static 메소드에서는 사용할 수 없습니다.
static 메소드는 객체가 인스턴스화되기 전에 먼저 메모리에 로드됩니다.
즉 클래스의 제네릭 타입에 의존할 수 없는것이죠.

따라서 메소드 자체의 제네릭 타입을 사용해야합니다.
```suggestion
@Component
public class GeneratorDeepCopiedType {

    public <K, V> Map<K, V> copiedMap(Map<K, V> map){
        Map<K, V> copiedMap = new ConcurrentHashMap<>();

        for(Map.Entry<K, V> entry : map.entrySet()){
            copiedMap.put(entry.getKey(), entry.getValue());
        }

        return copiedMap;
    }
}
```